### PR TITLE
Remove t.Skip in testcases

### PR DESCRIPTION
**Pull request by tnasu**
## Description

I fixed `t.Skip` and `t.Skipf` test cases.
```
.//consensus/state_test.go:     t.Skip("NON-deterministic test: no such LastProofHash making index validator to be proposer")
.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
.//types/voter_set_test.go:     t.Skip("this test case need a new reward rule")
.//types/validator_set_test.go: t.Skip("Temporarily excluded because this a case that doesn't end due to Proposer selection changes.")
.//types/validator_set_test.go: t.Skip("The ProposerPriority validated in this test case is excluded because it is not used in VRF.")
.//test/e2e/tests/e2e_test.go:          t.Skip("E2E_MANIFEST not set, not an end-to-end test run")
.//test/e2e/tests/validator_test.go:    t.Skip("NON-deterministic test: SelectProposer is not well since valSchedule.Set is changed")
.//state/state_test.go: t.Skip("This test is for priority based proposer. Vrf selection based proposer skips this test.")
.//state/state_test.go: t.Skip("Ostracon doesn't select a Proposer based on ProposerPriority")
.//p2p/transport_test.go:       t.Skip("NON-deterministic test: write tcp 127.0.0.1:40973->127.0.0.1:54732: i/o timeout")
.//mempool/reactor_test.go:             t.Skip("skipping test in short mode.")
.//mempool/reactor_test.go:             t.Skip("skipping test in short mode.")
.//light/detector_test.go:      t.Skip("Voter selection in Ostracon only supports sequential verification mode, " +
.//light/detector_test.go:      t.Skip("Voter selection in Ostracon only supports sequential verification mode, " +
.//light/client_test.go:        t.Skip("Skipping verification disabled under selection of voters")
```

```
.//consensus/replay_test.go:    t.Skip("We will skip this test case and prepare another one later" +
.//consensus/state_test.go:     t.Skipf("NON-deterministic test: race detected during execution of test")
.//libs/clist/clist_test.go:            t.Skipf("Skipping on non-amd64 machine")
.//libs/clist/clist_test.go:            t.Skipf("Skipping on non-amd64 machine")
```

### CODE REVIEW

**Comment by tnasu** [comment]
Addressed the below:

```
.//consensus/state_test.go:     t.Skip("NON-deterministic test: no such LastProofHash making index validator to be proposer")
TestStateLockPOLRelock

.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
TestHandshakeReplayAll

.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
TestHandshakeReplaySome

.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
TestHandshakeReplayOne

.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
TestHandshakeReplayNone

.//consensus/replay_test.go:    t.Skip("With decision to skip TestSimulateValidatorsChange, we also skip this test case" +
TestMockProxyApp

.//types/voter_set_test.go:     t.Skip("this test case need a new reward rule")
TestElectVotersNonDupEquity -> BenchmarkElectVotersNonDupEquity

.//types/validator_set_test.go: t.Skip("Temporarily excluded because this a case that doesn't end due to Proposer selection changes.")
TestValSetUpdatesOrderIndependenceTestsExecute

.//types/validator_set_test.go: t.Skip("The ProposerPriority validated in this test case is excluded because it is not used in VRF.")
TestAveragingInIncrementProposerPriority

.//test/e2e/tests/validator_test.go:    t.Skip("NON-deterministic test: SelectProposer is not well since valSchedule.Set is changed")
.//state/state_test.go: t.Skip("This test is for priority based proposer. Vrf selection based proposer skips this test.")
TestProposerFrequency

.//state/state_test.go: t.Skip("Ostracon doesn't select a Proposer based on ProposerPriority")
.//p2p/transport_test.go:       t.Skip("NON-deterministic test: write tcp 127.0.0.1:40973->127.0.0.1:54732: i/o timeout")
TestTransportHandshake

.//consensus/replay_test.go:    t.Skip("We will skip this test case and prepare another one later" +
TestSimulateValidatorsChange

.//consensus/state_test.go:     t.Skipf("NON-deterministic test: race detected during execution of test")
TestResetTimeoutPrecommitUponNewHeight
```

Keep `t.Skip` and `t.Skipf`
```
//
// Skip
//
.//test/e2e/tests/e2e_test.go:          t.Skip("E2E_MANIFEST not set, not an end-to-end test run")
.//mempool/reactor_test.go:             t.Skip("skipping test in short mode.")
.//mempool/reactor_test.go:             t.Skip("skipping test in short mode.")

//
// Skipf
//
.//libs/clist/clist_test.go:            t.Skipf("Skipping on non-amd64 machine")
.//libs/clist/clist_test.go:            t.Skipf("Skipping on non-amd64 machine")

//
// depends on the deprecated `Skipping verification` mode of Ostracon
//
.//light/detector_test.go:      t.Skip("Voter selection in Ostracon only supports sequential verification mode, " +
.//light/detector_test.go:      t.Skip("Voter selection in Ostracon only supports sequential verification mode, " +
.//light/client_test.go:        t.Skip("Skipping verification disabled under selection of voters")

```

File: consensus/common_test.go [hunk 1]
```diff
@@ -195,19 +195,8 @@ func decideProposal(
...
 ) (proposal *types.Proposal, block *types.Block) {
-	oldPrivValidator := cs1.privValidator
-	oldPrivValidatorPubKey := cs1.privValidatorPubKey
 	cs1.mtx.Lock()
-	pubKey1, _ := cs1.privValidator.GetPubKey()
-	pubKey2, _ := vs.PrivValidator.GetPubKey()
-	if !pubKey1.Equals(pubKey2) {
-		// block creator must be the cs.privValidator
-		cs1.privValidator = vs.PrivValidator
-		cs1.privValidatorPubKey = pubKey2
```
Comment (tnasu) [inline consensus/common_test.go:206]: These technics are happened `deadlock` by `cs1.createProposalBlock(round)` with `test --race`. And I made `createProposalBlockSlim`.

File: state/state_test.go [hunk 2]
```diff
@@ -494,24 +492,67 @@ func testProposerFreq(t *testing.T, caseNum int, valSet *types.ValidatorSet) {
...
+			require.True(
+				t,
+				abs <= bound,
+				fmt.Sprintf("Case %d val %d (%d): got %d, expected %d", caseNum, i, N, gotFreq, expectFreq),
+			)
+		}
+	*/
+
+	// Chi-squared test for `SelectProposer`
+	chiSquareds := make([]ChiSquared, N)
```
Comment (tnasu) [inline state/state_test.go:518]: I replaced `frequencies test(bound: margin of error)` with `Chi-squared test`

**Review by torao — APPROVED** [review]
LGTM

**Review by kokeshiM0chi — APPROVED** [review]
LGTM

**Pull request merged**